### PR TITLE
Support table parameters

### PR DESCRIPTION
**Pull request by max-jacobs**
Making a small update for SY to support the latest Circus Train and to allow skipping CT's housekeeping process.

Main change is in `CircusTrainRunner`

### CODE REVIEW

File: shunting-yard-replicator/src/main/java/com/expediagroup/shuntingyard/replicator/exec/conf/OrphanedDataStrategyConfig.java [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+package com.expediagroup.shuntingyard.replicator.exec.conf;
+
+import org.springframework.boot.context.properties.ConfigurationProperties;
+import org.springframework.context.annotation.Configuration;
+
+import com.hotels.bdp.circustrain.api.conf.OrphanedDataStrategy;
+
+@Configuration
+@ConfigurationProperties
+public class OrphanedDataStrategyConfig {
```
Comment (massdosage) [inline shunting-yard-replicator/src/main/java/com/expediagroup/shuntingyard/replicator/exec/conf/OrphanedDataStrategyConfig.java]: OK, so we need a suffix on this one since it's not the actual `OrphanedDataStrategy`. I see we have one other class in this package called `EventReceiverConfiguration` - should we follow the naming convention and change `Config`->`Configuration`?

File: shunting-yard-replicator/src/main/java/com/expediagroup/shuntingyard/replicator/exec/conf/OrphanedDataStrategyConfig.java [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+ */
+package com.expediagroup.shuntingyard.replicator.exec.conf;
+
+import org.springframework.boot.context.properties.ConfigurationProperties;
+import org.springframework.context.annotation.Configuration;
+
+import com.hotels.bdp.circustrain.api.conf.OrphanedDataStrategy;
+
+@Configuration
+@ConfigurationProperties
```
Comment (massdosage) [inline shunting-yard-replicator/src/main/java/com/expediagroup/shuntingyard/replicator/exec/conf/OrphanedDataStrategyConfig.java]: Doesn't this need a prefix "orphaned-data-strategy" to get bound to the value from the YAML?

File: README.md [hunk 3]
```diff
@@ -147,6 +164,7 @@ The table below describes all the available configuration values for Shunting Ya
 |`sqs.queue`|Yes|Fully qualified URI of the [AWS SQS](https://aws.amazon.com/sqs/) Queue to read the Hive events from.|
 |`sqs.wait.time.seconds`|No|Wait time in seconds for which the receiver will poll the SQS queue for a batch of messages. Default is 10 seconds. Read more about long polling with AWS SQS [here](https://docs.aws.amazon.com/AWSSimpleQueueService/latest/SQSDeveloperGuide/sqs-long-polling.html).|
 |`source-table-filter.table-names`|No|A list of tables selected for Shunting Yard replication. Supported format: `database_1.table_1, database_2.table_2`. If these are not provided, Shunting Yard will not replicate any tables.|
+|`orphaned-data-strategy`|No|Orphaned data strategy to use for replications. Possible values: `NONE` and `HOUSEKEEPING`. Default is `HOUSEKEEPING`.|
```
Comment (massdosage) [inline README.md]: I'm missing where this default happens in the code? I see you've added the ability to set an orphaned data strategy but is there a test (maybe an existing one so it doesn't show up in the PR) that shows that if you don't set this then the value is `HOUSEKEEPING`?

File: shunting-yard-replicator/src/main/java/com/expediagroup/shuntingyard/replicator/exec/conf/OrphanedDataStrategyConfig.java (hunk 1 above)
Comment (max-jacobs) [reply shunting-yard-replicator/src/main/java/com/expediagroup/shuntingyard/replicator/exec/conf/OrphanedDataStrategyConfig.java]: Yes good point, one day I'll name these config classes appropriately...

File: shunting-yard-replicator/src/main/java/com/expediagroup/shuntingyard/replicator/exec/conf/OrphanedDataStrategyConfig.java (hunk 2 above)
Comment (max-jacobs) [reply shunting-yard-replicator/src/main/java/com/expediagroup/shuntingyard/replicator/exec/conf/OrphanedDataStrategyConfig.java]: I don't _think_ so because in this case there are no "sub" properties. Spring is clever enough to know to map the yaml property "orphaned-data-strategy" to the field in here called "orphanedDataStrategy". I've tested this out on EMR doing an actual replication and it works. Shame there's no integration tests in SY like in CT to test out different config settings.

File: README.md (hunk 3 above)
Comment (max-jacobs) [reply README.md]: this happens in `OrphanedDataStrategyConfig` line 26 the default is if no yaml configuration overrides this initial setting. 

So there's no test for this because i guess it would require an integration test of sorts in order to test the mapping from yaml to bean, and we don't have these in SY at the moment. should I make it part of this PR to add these?

File: shunting-yard-replicator/src/test/java/com/expediagroup/shuntingyard/replicator/exec/external/CircusTrainConfigTest.java [hunk 4]
```diff
@@ -96,6 +96,7 @@ public void withMultiValuePartitionFilter() {
     assertThat(replication.getReplicaTable().getDatabaseName()).isEqualTo("replicaDatabaseName");
     assertThat(replication.getReplicaTable().getTableName()).isEqualTo("replicaTableName");
     assertThat(replication.getReplicaTable().getTableLocation()).isEqualTo("replicaTableLocation");
+    assertThat(replication.getOrphanedDataStrategy()).isEqualTo(OrphanedDataStrategy.HOUSEKEEPING);
```
Comment (abhimanyugupta07) [inline shunting-yard-replicator/src/test/java/com/expediagroup/shuntingyard/replicator/exec/external/CircusTrainConfigTest.java]: Will be good to add a test when `OrphanedDataStrategy` is null.

The `CircusTrainConfig` doesn't do a null check on this argument at the moment.

File: shunting-yard-replicator/src/main/java/com/expediagroup/shuntingyard/replicator/exec/conf/OrphanedDataStrategyConfig.java (hunk 2 above)
Comment (patduin) [reply shunting-yard-replicator/src/main/java/com/expediagroup/shuntingyard/replicator/exec/conf/OrphanedDataStrategyConfig.java]: I think can be separate we already have an issue: https://github.com/ExpediaGroup/shunting-yard/issues/5

File: README.md [hunk 5]
```diff
@@ -135,6 +135,23 @@ In this case, the replica database name is not provided in the `table-replicatio
...
+    table-replications:
+      - source-table:
+          database-name: source_database
+          table-name: test_table
+        replica-table:
+          database-name: replica_database
+          table-name: test_table_1 
+    orphaned-data-strategy: NONE 
+ 
+This will ensure that Shunting Yard only starts up Circus Train's replication module, and that no paths are added to a Housekeeping database.
```
[the remark this answers is not in the document]
Comment (max-jacobs) [reply README.md]: Yes may be good to provide some context, but Beekeeper isn't a "strategy" that's available as an options - only "none" and "housekeeping".  Something like this:

> This is a necessary step if you want to make use of an external housekeeping mechanism, for example [Beekeeper](https://github.com/ExpediaGroup/beekeeper).

?

File: README.md [hunk 6]
```diff
@@ -163,6 +181,18 @@ Graphite configuration can be passed to Shunting Yard using an optional `--ct-co
       namespace: com.company.shuntingyard
       prefix: dev
 
+### Configuring table transformations
```
[the remark this answers is not in the document]
Comment (max-jacobs) [reply README.md]: Will update this

File: README.md (hunk 5 above)
[the remark this answers is not in the document]
Comment (abhimanyugupta07) [reply README.md]: Sounds good. I have updated my suggestion.

File: shunting-yard-replicator/src/test/java/com/expediagroup/shuntingyard/replicator/exec/external/CircusTrainConfigTest.java (hunk 4 above)
Comment (max-jacobs) [reply shunting-yard-replicator/src/test/java/com/expediagroup/shuntingyard/replicator/exec/external/CircusTrainConfigTest.java]: Good point, will add this in, thanks

File: README.md (hunk 5 above)
[the remark this answers is not in the document]
Comment (max-jacobs) [reply README.md]: Needs a comma after "data" not a full stop.

[the remark this answers is not in the document]
Comment (max-jacobs) [reply README.md]: I've managed to update and commit, thanks!

**Comment by max-jacobs** [comment]
This now uses the latest Circus Train version, also tested this on EMR and it works, so going to merge.
